src/main/java/io/openliberty/tools/intellij/LibertyExplorer.java

```diff
@@ -48,23 +51,37 @@ public class LibertyExplorer extends SimpleToolWindowPanel {
 
     public LibertyExplorer(@NotNull Project project) {
         super(true, true);
-        // build tree
-        Tree tree = buildTree(project, getBackground());
-
-        if (tree != null) {
-            JBScrollPane scrollPane = new JBScrollPane(tree);
-            scrollPane.setName(Constants.LIBERTY_SCROLL_PANE);
-            this.setContent(scrollPane);
-        } else {
-            JBTextArea jbTextArea = new JBTextArea(LocalizedResourceUtil.getMessage("no.liberty.projects.detected"));
-            jbTextArea.setEditable(false);
-            jbTextArea.setBackground(getBackground());
-            jbTextArea.setLineWrap(true);
-
-            this.setContent(jbTextArea);
-        }
-        ActionToolbar actionToolbar = buildActionToolbar(tree);
-        this.setToolbar(actionToolbar.getComponent());
+        ModalityState modalityState = getModalityState();
+        ApplicationManager.getApplication().executeOnPooledThread(() -> {
```

Comment:
It would be a good idea to add some documentation here which describes the problem being solved and the solution, in particular if this was suggested by JetBrains documentation or some other source it would be helpful to provide a link to that guidance.

Reply:
I will add documentation as comments. Jetbrains Documentation link https://plugins.jetbrains.com/docs/intellij/general-threading-rules.html that I referred

Reply:
Looks good now. Thanks.